chimedb/core/orm.py

```diff
@@ -296,7 +298,22 @@ def connect_database(read_write=False, reconnect=False):
 
     from . import connectdb
 
-    connectdb.connect(reconnect)
+    # Try multiple time to connect
+    for i in range(ntries):
+        try:
+            connectdb.connect(reconnect)
+        except ConnectionError:
+            if i == ntries - 1:
+                logger.warn(f"Failed to connect after {ntries} attempts.")
+                raise
+            else:
+                logger.log(
+                    f"Failed to connect attempt {i+1} of {ntries}. "
+                    f"Trying again in {60} seconds."
+                )
+                time.sleep(60)
+        else:
+            break
```

Comment:
A few suggestions to clean up the logic slightly (i.e. remove the slightly obscure `try...except...else` construct), and also just raise the final error rather than warning about it.

Also removed the unnecessary `{60}` format

```suggestion
        try:
            connectdb.connect(reconnect)
            break
        except ConnectionError as e:
            if i == ntries - 1:
                raise ConnectionError(
                    f"Failed to connect to chimedb after {ntries} attempts."
                ) from e

            logger.info(
                f"Failed to connect attempt {i+1} of {ntries}. Trying again in 60 seconds."
            )
            time.sleep(60)
```